src/io/leopard/toLeopard.ts

```diff
@@ -663,149 +1155,375 @@ export default function toLeopard(
+          if (desiredInputShape === "index") {
+            if ((block.inputs.NUM2 as BlockInput.Any).type !== "block" && !isNaN(Number(block.inputs.NUM2.value))) {
+              satisfiesInputShape = "index";
+              if (Number(block.inputs.NUM2.value) === 1) {
+                blockSource = `(${inputToJS(block.inputs.NUM1, "number")})`;
+              } else {
+                blockSource = `((${inputToJS(block.inputs.NUM1, "number")}) + ${(block.inputs.NUM2.value as number) -
+                  1})`;
+              }
+              break;
+            } else if (
+              (block.inputs.NUM1 as BlockInput.Any).type !== "block" &&
+              !isNaN(Number(block.inputs.NUM1.value))
+            ) {
+              satisfiesInputShape = "index";
+              if (Number(block.inputs.NUM1.value) === 1) {
+                blockSource = `(${inputToJS(block.inputs.NUM2, "number")})`;
+              } else {
+                blockSource = `(${(block.inputs.NUM2.value as number) - 1} + ${inputToJS(
+                  block.inputs.NUM2,
+                  "number"
+                )})`;
+              }
+              break;
+            }
+          }
+          satisfiesInputShape = "number";
+          blockSource = `((${inputToJS(block.inputs.NUM1, "number")}) + (${inputToJS(block.inputs.NUM2, "number")}))`;
+          break;
+
         case OpCode.operator_subtract:
-          return `((${inputToJS(block.inputs.NUM1)}) - (${inputToJS(block.inputs.NUM2)}))`;
+          satisfiesInputShape = "number";
+          blockSource = `((${inputToJS(block.inputs.NUM1, "any")}) - (${inputToJS(block.inputs.NUM2, "any")}))`;
+          break;
+
         case OpCode.operator_multiply:
-          return `((${inputToJS(block.inputs.NUM1)}) * (${inputToJS(block.inputs.NUM2)}))`;
+          satisfiesInputShape = "number";
+          blockSource = `((${inputToJS(block.inputs.NUM1, "any")}) * (${inputToJS(block.inputs.NUM2, "any")}))`;
+          break;
+
         case OpCode.operator_divide:
-          return `((${inputToJS(block.inputs.NUM1)}) / (${inputToJS(block.inputs.NUM2)}))`;
+          satisfiesInputShape = "number";
+          blockSource = `((${inputToJS(block.inputs.NUM1, "any")}) / (${inputToJS(block.inputs.NUM2, "any")}))`;
+          break;
+
         case OpCode.operator_random:
-          return `this.random(${inputToJS(block.inputs.FROM)}, ${inputToJS(block.inputs.TO)})`;
+          satisfiesInputShape = "number";
+          blockSource = `this.random(${inputToJS(block.inputs.FROM, "any")}, ${inputToJS(block.inputs.TO, "any")})`;
+          break;
+
         case OpCode.operator_gt:
-          return `((${inputToJS(block.inputs.OPERAND1)}) > (${inputToJS(block.inputs.OPERAND2)}))`;
+          satisfiesInputShape = "boolean";
+          blockSource = `(this.compare((${inputToJS(block.inputs.OPERAND1, "any")}), (${inputToJS(
+            block.inputs.OPERAND2,
+            "any"
+          )})) > 0)`;
```

Comment:
Worth investigating if it's possible to skip `this.compare` in some cases here, particularly if one of the inputs is a number.